documentation/content/Contributing/automated-release.md

```diff
@@ -1,42 +1,93 @@
 ---
-Title: Fully automated version management and publishing
-Date: 2019-07-20 23:17
+Title: Automated Version Management and Publishing
+Subtitle:
 Slug: fully-automated-version-management-and-publishing
 Category: Contributing
-Authors: Talha Mansoor
+Tags:
+Date: 2019-07-20 23:17
+Summary: Elegant utilizes a 'release early, release often' philosophy that embraces a fully automated release process.
+Keywords:
+Authors: Talha Mansoor, Jack De Winter
 ---
 
-Elegant follows "release early, release often" software development philosophy. In this regard, our release process is fully automated.
+[TOC]
+
+The Elegant project follows a "release early, release often" software development philosophy.
+By employing this philosophy, we are able to have each team member working on a different
+aspect of the project with little friction from each other's changes. At any point, any team
+member can pull changes that have been made in another branch to their fork of the repository
+and exercise their changes with complete confidence.
+
+Another benefit of this philosophy if that in order to attain this goal, our release process
+must be fully automated.
+
+## Release Versions
+
+Version management and publishing is accomplished using the
+[semantic-release](https://github.com/semantic-release/semantic-release) tool and it's
+[configuration file](https://github.com/Pelican-Elegant/elegant/blob/master/.releaserc.json).
+This tool entirely removes any human intervention needed for determining how to label
+the next release of the project.
+
+## CHANGELOG File Updates
+
+An integral part of the release is the generation of information to be added to the
+[CHANGELOG file](https://github.com/Pelican-Elegant/elegant/blob/master/CHANGELOG.md).
+By mandating that any commits for the project follow a
+[mandated format for the commit messages]({filename}./git-commit-guidelines.md),
+the commits being added to the release can have their commit messages scanned by a tool,
+including relevant portions into the release notes. Similar to the previous paragraph on
```

Comment:
> Similar to the previous paragraph on
version management, this can be accomplished without any need for human intervention.

This too is done automatically.

Reply:
done.